cookbooks/aws-parallelcluster-slurm/recipes/enable_chef_error_handler.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+#
+# Cookbook:: aws-parallelcluster
+# Recipe:: enable_chef_error_handler
+#
+# Copyright:: 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+chef_handler 'WriteChefError::SlurmChefHandler' do
```

Comment:
Question: It is not clear to me why do we only add the handler under slurm recipes, what if the failure happen outside Slurm recipes?

Reply:
This handler is designed to be slurm-specific, because the error json file contains a lot of slurm-specific information (such as slurm node name, etc). So it should only be registered when the scheduler is slurm. As long as the sheduler is slurm and this hander get registered, then it will handle any error during the run, regardless of where the failure comes from, Slurm recipes, or other recipes.